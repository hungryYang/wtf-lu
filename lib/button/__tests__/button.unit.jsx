import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../button';
import {mount} from 'enzyme';

describe('按钮测试', () => {
  it('普通模式', () => {
    const tree = renderer
      .create(<Button className="test">测试</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('带icon', () => {
    const tree = renderer
      .create(<Button className="test" icon="add">测试</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('icon位置测试', () => {
    const tree = renderer
      .create(<Button className="test" icon="add" iconPosition="right">测试</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  // it('按钮点击测试', () => {
  //   const fn = jest.fn()
  //   const wrapper = mount(<Button icon="add" onClick={fn}/>);
  //   wrapper.find('svg').simulate('click')
  //   expect(fn).toBeCalled();
  // });
});


