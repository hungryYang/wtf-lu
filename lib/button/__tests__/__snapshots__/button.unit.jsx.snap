// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`按钮测试 icon位置测试 1`] = `
<button
  className="wtf-button test"
>
  测试
   
  <svg
    className="wtf-icon"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
</button>
`;

exports[`按钮测试 带icon 1`] = `
<button
  className="wtf-button test"
>
  <svg
    className="wtf-icon"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
   
  测试
</button>
`;

exports[`按钮测试 普通模式 1`] = `
<button
  className="wtf-button test"
>
   
  测试
</button>
`;
